src/rate-limiter.ts

```diff
@@ -20,8 +20,9 @@ export class RateLimiter {
     }
 
     public checkForLimiting = (httpResponse: MinimalHTTPResponse): void => {
+        let text: string | undefined
         try {
-            const text = httpResponse.responseText
+            text = httpResponse.responseText
```

Comment:
Actually - why not just moving the declaration up? It's a totally safe access and doesn't need to be in the try block